src/WebJobs.Extensions.NotificationHubs/Bindings/NotificationHubAsyncCollector.cs

```diff
@@ -4,23 +4,47 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.NotificationHubs;
+using Microsoft.Azure.WebJobs.Host;
 
 namespace Microsoft.Azure.WebJobs.Extensions.NotificationHubs
 {
     internal class NotificationHubAsyncCollector : IAsyncCollector<Notification>
     {
         private INotificationHubClientService _notificationHubclientService;
         private string _tagExpression;
+        private bool _enableTestSend;
+        private TraceWriter _traceWriter;
 
-        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression)
+        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression, bool enableTestSend, TraceWriter traceWriter)
         {
             _notificationHubclientService = clientService;
             _tagExpression = tagExpression;
+            _enableTestSend = enableTestSend;
+            _traceWriter = traceWriter;
         }
 
         public async Task AddAsync(Notification item, CancellationToken cancellationToken = default(CancellationToken))
         {
-            await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            NotificationOutcome notificationOutcome = await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            if (_enableTestSend)
+            {
+                string debugLog = $"NotificationHubs Test Send\r\n" +
+                    $"TrackingId = {notificationOutcome.TrackingId}\r\n" +
+                    $"State = {notificationOutcome.State}\r\n" +
+                    $"Results (Success = {notificationOutcome.Success}, Failure = {notificationOutcome.Failure})\r\n";
+                if (notificationOutcome.Results == null || notificationOutcome.Results.Count == 0)
+                {
+                    debugLog += $"Results are null or empty. Refer to NotificationHubs documentaion for more details on how to debug issues while sending push notifications.";
+                }
+                else
```

Comment:
I'd remove this log. It'll already show that they have 0 success and 0 failure and if they've turned this on, they likely know they can look at the docs. It'll be cleaner without this. So just write out the details below if Results != null